spec/controllers/course_subjects_controller_spec.rb

```diff
@@ -0,0 +1,89 @@
+require "rails_helper"
+
+RSpec.describe CourseSubjectsController, type: :controller do
+  let(:user){FactoryBot.create :user, role_id: "supervisor"}
+  let(:user2){FactoryBot.create :user2, role_id: "trainee"}
+  let(:course){FactoryBot.create :course}
+  let(:subject){FactoryBot.create :subject}
+  let(:course_subject){FactoryBot.create :course_subject, course: course,
+                        subject: subject}
+
+  before{sign_in user}
```

Comment:
dùng before all dc k

Reply:
Có vẻ không được ạ. Bản chất của let là tạo ra đối tượng dùng cho từng example (khối it), còn before(:all) hay before(:context) dùng để khai báo môi trường dùng cho cả context lớn. Mà user được khai báo trong let nên có vẻ chúng xung đột nhau và không dùng được ạ.